plugins/org.python.pydev/tests_analysis/com/python/pydev/analysis/OccurrencesAnalyzer2Test.java

```diff
@@ -846,4 +846,27 @@ public void testInvalidEncoding() throws Exception {
     //        );
     //        checkError(1);
     //    }
+
+    public void testNoSelfInZopeInterface() throws Exception {
+        doc = new Document(""
+                + "class I(Interface):\n"
+                + "    def F():\n"
+                + "        \"\"\"Doc\"\"\"\n"
+                + "@implementer(I)\n"
+                + "class Ic:\n"
+                + "    def F(self):\n"
+                + "        print \"->\", self, \"Object\"\n"
+                + "c = Ic()\n"
+                + "print type(I), type(Ic), type(c), type(Ic.F), type(c.F)\n"
+                + "c.F()");
+        checkError("Undefined variable: Interface", "Undefined variable: implementer");
+    }
+
+    public void testNoSelfInSomeInterface() throws Exception {
+        doc = new Document(""
+                + "class I('Interface'):\n"
```

Comment:
This test doesn't seem correct... I don't think you can really inherit from a string...